### DISCUSSION TITLE
How to copy a page into a PdfStamper?

### DISCUSSION THREAD

**Discussion by jon-m-craig** [LibrePDF/OpenPDF#1038] [Q&A]
So I have a PDF template of a pre-printed form (it's a medical insurance form) - it has fields for everything.  I use a `PdfStamper` to fill out the fields - and all is fine.

Sometimes, though, I need to add a page because the form needs another copy of one of the pages.

So I'm doing,

```
            stamper.insertPage(stamper.getReader().getNumberOfPages() + 1, stamper.getReader().getPageSize(1));
            stamper.replacePage(stamper.getReader(), 1, stamper.getReader().getNumberOfPages() + 1);
```

This seems like it would copy page 1 from the `reader` into the just-newly-created page x+1 in the `stamper` and all would be fine.  Problem is, instead it throws an exception:

```
java.lang.NullPointerException: Cannot invoke "com.lowagie.text.pdf.PdfDictionary.remove(com.lowagie.text.pdf.PdfName)" because "dic2" is null
	at com.lowagie.text.pdf.PdfStamperImp.replacePage(PdfStamperImp.java:736)
	at com.lowagie.text.pdf.PdfStamper.replacePage(PdfStamper.java:203)
```

I'm not sure what the issue is - there's definitely a page 1 in the reader, so I'm not sure why anything is null.

Maybe I'm going about the page copy in the wrong way; is there a suggested/better way to handle it?

**Comment by andreasrosdalw** [comment]
This is the code where you get a NullPointerException:
https://github.com/LibrePDF/OpenPDF/blame/master/openpdf/src/main/java/com/lowagie/text/pdf/PdfStamperImp.java#L736

Can you please study this code in the replacePage() method, and possibly this code can be improved by adding a check if the dic2 variable is null. Next can you test this change, and possibly make a pull request to fix this problem?

**Reply by jon-m-craig** [reply]
Uh, that's way above my head.  I'm merely trying to use the library. :/
